server/core/src/repl/mod.rs

```diff
@@ -363,11 +368,60 @@ async fn repl_run_consumer(
     Some(socket_addr)
 }
 
+#[instrument(level="info", skip(idms), fields(eventid=Uuid::new_v4().to_string()))]
+async fn replication_trigger(
+    idms: &IdmServer,
+    last_replication_instant: &Instant,
+    enable_experimental_replication_trigger: bool,
+    replication_interval: Duration,
+) {
+    tokio::select! {
+        _ = async {
+            let mut read_txn = idms.proxy_read().await;
+
+            let mut one_sec_int = interval(Duration::from_secs(1));
+            // sleep for the minimum window size before we do anything
+            sleep(Duration::from_secs(MIN_REPL_TASK_POLL_INTERVAL)).await;
+            loop {
+                let write_ops_since_last_repl = read_txn.qs_read.get_write_ops_since_last_repl();
+                let time_since_last_repl = last_replication_instant.elapsed().as_secs();
+
+                let Some(write_throughput_index) = time_since_last_repl.checked_sub(MIN_REPL_TASK_POLL_INTERVAL) else {
+                    error!("Negative replication window index detected, aborting...");
+                    return;
```

Comment:
We should handle this error better than returning, because a return here will kill the repl task I think?

Reply:
Just do saturating sub instead.

Reply:
> We should handle this error better than returning, because a return here will kill the repl task I think?

I mean I haven't tested it but I assumed it would just immediately return the future? I'd expect the repl loop to resume normally. Actaully if returning kills the entire repl task, then also this [return](https://github.com/Seba-T/kanidm/blob/53a866bdc0a8b6d47c5aa7c8a1f7a3239df3b12e/server/core/src/repl/mod.rs#L414) should kill it, but I'm pretty sure it doesn't since I've tested it.